packages/router/src/adapters/cache/auction.ts

```diff
@@ -0,0 +1,54 @@
+import { BigNumber } from "ethers";
+
+import { CachedBid } from "../../lib/entities/cache";
+
+// TODO: This could use some optimization.
+// - For one, we should be storing via Redis (and move this to adapters) to keep the state persistent across restarts.
+// - Store by transactionId in a nested Map (instead of an Array)?
+// - We could also optimize by comparing user asks to actual funding in their wallet, and only consider
+//   promised liquidity actually outstanding if the user *can* go through with the bid.
+/**
+ * Bid caching helper class to ensure we only store one bid per assetId/chainId combination, clean up expired bids,
+ * and provide a quick calculation endpoint to determine outstanding liquidity.
+ */
+export class AuctionCache {
```

Comment:
Wondering the best way to handle the case where we win the bid, and the bid expires before we are able to prepare the receiver. Perhaps there should be a way to increase the expiry or set a `confirmed` flag?

Reply:
oh right, like we want to keep the liquidity reserved, right?